src/include/catalog/postgres/nameBuilder.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <string>
+
+#include "catalog/postgres/pg_defs.h"
+
+namespace terrier::catalog::postgres {
+/* NameBuilder generates names for database objects such as constraints and indexes
+ *
+ * Following Postgres naming convention
+ * */
+class NameBuilder {
+ public:
+  /*
+   * Construct an object name
+   * @param table_name name of the table to get foreign key constraints
+   * @param field_name name of the field
+   * @param type name of the type
+   * @return name of the object
+   * */
+  static std::string MakeName(const std::string &table_name, const std::string &field_name, const std::string &type) {
```

Comment:
Replace 'type' with a `enum class` so that we can do compile time checks. You will need a switch statement in this function that converts the enum to the proper string.